helix-core/src/syntax.rs

```diff
@@ -869,7 +882,8 @@ impl Syntax {
             ts_parser.cursors.push(cursor);
 
             // Remove all untouched layers
-            self.layers.retain(|id, _| touched.contains(&id));
+            self.layers
+                .retain(|_, layer| replace(&mut layer.flags, 0) & LAYER_TOUCHED != 0);
```

Comment:
Don't we need to reset these flags at the end/start of the loop?

Reply:
I am not really sure what yoy mean. The `replace(.., 0)` does reset the flags here and retain calls the closure for every layer so we do end up resisting all flags.

It can not be done earlier then the retain because we still need to read the `touched` fiags here.

Reply:
Ah sorry, I missed that 👍